enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+       type: AWS
+ status:
+   componentRoutes:
+```
+3. Ingress Operator will create the default ingress controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
+#### Create an Ingress Controller using Ingress Controller CR and assign the EIP allocation mentioned in the Ingress Controller CR.
+1. Cluster administrator creates an Ingress Controller:
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  creationTimestamp: null
+  name: test
+  namespace: openshift-ingress-operator
+spec:
+  endpointPublishingStrategy:
+    loadBalancer:
+      scope: External
+      providerParameters:
+        type: AWS
+        aws:
+          type: NLB
+          networkLoadBalancer:
+            eip-allocations: eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>
+    type: LoadBalancerService
+```
+
+Ingress Operator will create an NLB Ingress Controller called `test` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+#### Cluster administrator adds eip-allocations field the existing ingress controller
+
+Ingress Operator will create an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+#### Cluster administrator updates eip-allocations field the existing ingress controller
+
+Ingress Operator will update an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+### API Extensions
+- The API for setting `eip-allocations` field in the installer-config will look as follows in the file `pkg/types/aws/platform.go`:
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+...
+    // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb 
+    // +optional 
+    LBType configv1.AWSLBType `json:"lbType,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *configv1.AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+```
+
+- The API for the `eip-allocations` field in the ingress cluster object to store the value of the
+  field `eip-allocations` from the installer which the ingress operator uses to create the ingress controller
+  will look as follows in the file `config/v1/types_ingress.go`:
```

Comment:
```suggestion
- Change the AWSIngressSpec type in the file `config/v1/types_ingress.go`:
```

Reply:
But this does not explain the process.

Reply:
The wording was not clear.  Maybe something like "The installer will propagae the EIP allocations to the cluster ingress configuration"?

Reply:
i modified it to ` The installer will propagate the eipAllocations to the cluster ingress configuration which the ingress operator uses to create the default Ingress Controller
  will look as follows in the file `config/v1/types_ingress.go` of `openshift/api` github repo:`